drivers/spi/spi_bitbang.c

```diff
@@ -88,8 +79,50 @@ static int spi_bitbang_transceive(const struct device *dev,
 	const struct spi_bitbang_config *info = dev->config;
 	struct spi_bitbang_data *data = dev->data;
 	struct spi_context *ctx = &data->ctx;
+	int rc;
+	const struct gpio_dt_spec *miso = NULL;
+	const struct gpio_dt_spec *mosi = NULL;
 
-	spi_bitbang_configure(info, data, spi_cfg);
+	rc = spi_bitbang_configure(info, data, spi_cfg);
+	if (rc < 0) {
+		return rc;
+	}
+
+	if (spi_cfg->operation & SPI_HALF_DUPLEX) {
+		if (tx_bufs && rx_bufs) {
+			LOG_ERR("Both RX and TX specified in half duplex mode");
+			return -EINVAL;
+		} else if (tx_bufs && !rx_bufs) {
+			/* TX mode */
+			rc = gpio_pin_configure_dt(&info->mosi_gpio,
+						   GPIO_OUTPUT_INACTIVE);
+			mosi = &info->mosi_gpio;
+		} else if (!tx_bufs && rx_bufs) {
+			/* RX mode */
+			rc = gpio_pin_configure_dt(&info->mosi_gpio,
+						   GPIO_INPUT);
+			miso = &info->mosi_gpio;
+		}
+
+		if (rc < 0) {
+			LOG_ERR("Couldn't configure MOSI pin: %d", rc);
+			return rc;
+		}
+	} else {
+		if (info->mosi_gpio.port) {
+			mosi = &info->mosi_gpio;
+			rc = gpio_pin_configure_dt(&info->mosi_gpio,
+						   GPIO_OUTPUT_INACTIVE);
+			if (rc < 0) {
+				LOG_ERR("Couldn't configure MOSI pin: %d", rc);
+				return rc;
+			}
```

Comment:
By using a variable initialized to either `GPIO_INPUT` or `GPIO_OUTPUT_INACTIVE`, you could move this part after `if (spi_cfg->operation & SPI_HALF_DUPLEX)` and make it common for all the three paths. This would simplify the code a bit.

Reply:
That's a very good point. Fixed.